src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationSource.cs

```diff
@@ -11,21 +13,44 @@ internal class AzureAppConfigurationSource : IConfigurationSource
         private readonly bool _optional;
         private readonly Func<AzureAppConfigurationOptions> _optionsProvider;
         private readonly IConfigurationClientFactory _configurationClientFactory;
+        private readonly string _environmentName;
 
-        public AzureAppConfigurationSource(Action<AzureAppConfigurationOptions> optionsInitializer, bool optional = false, IConfigurationClientFactory configurationClientFactory = null)
+        public AzureAppConfigurationSource(Action<AzureAppConfigurationOptions> optionsInitializer, string environmentName = null, bool optional = false, IConfigurationClientFactory configurationClientFactory = null)
```

Comment:
NIT: I'd prefer we don't set default values for the parameters (environmentName and optional). It forces the caller to the set them explicitly. It's easier to read the code.

Reply:
I don't see the default value of `configurationClientFactory` is used anywhere.